addons/sourcemod/scripting/shavit-core.sp

```diff
@@ -1436,6 +1439,15 @@ void VelocityChanges(int data)
 	{
 		TeleportEntity(client, NULL_VECTOR, NULL_VECTOR, fAbsVelocity);
 	}
+
+	if (GetTimerStatus(client) == view_as<int>(Timer_Running))
+	{
+		float curVel = SquareRoot(Pow(fAbsVelocity[0], 2.0) + Pow(fAbsVelocity[1], 2.0));
+		float maxVel = gA_Timers[client].fMaxVelocity;
+		gA_Timers[client].fMaxVelocity = (curVel > maxVel) ? curVel : maxVel;
+		// STOLEN from Epic/Disrevoid. Thx :)
+		gA_Timers[client].fAvgVelocity += (curVel - gA_Timers[client].fAvgVelocity) / gA_Timers[client].iJumps;
+	}
```

Comment:
Implement the average as a sum / total frames. That way styles that don't jump can still benefit from an average velocity counter.